### DISCUSSION TITLE
Very long response times

### DISCUSSION THREAD

**Discussion by sebix** [komoot/photon#759] [General]
Dear team,

On our local test instance with 8 cores, 50GB of RAM (but only < 10 GB are used by photon) we observe a mean response time (all for the same country) of **> 2s**, with outliers < 1 minute.
In contrast, photon.komoot.io takes about **100ms** per request, which already includes network transport delays.

I have read https://github.com/komoot/photon/discussions/726 but found no significant improvements by setting different `-Xmx` and `-Xms` settings. What else can we improve? Could the HDD be the only reason?

Thank you for your help

**Comment by lonvia** [comment]
The combination of little RAM and HDD is the most likely contributing factor. Note that the planet is now 160GB. You'd want to keep most of the data cached in RAM and the rest needs to load fast.

photon.komoot.io currently runs on a machine with 128GB RAM and NVME SSD. I also note that the java on our instance reports in `top` that it is using 200GB of virtual RAM with 92GB being actually resident. So if your photon reports to be using less than 10GB, this seems odd. This is running on a out-of-the-box Debian 12 with no special parameters (`java -jar photon.jar -cors-any`) using the jar from github and the extract from the graphhopper servers.

**Reply by sebix** [reply]
This is very encouraging. Thank you very much for the details.
My test setup is also an OOTB Debian 12, just without SSD, as I don't have a setup with the combination of SSD and high RAM at hand. When only querying addresses in Germany, the RAM is only filled with about 3-4GB (which is actually about the data size for the data of Germany), but with high response times. So the issue must be really the missing SSD, I suspect.
